### ISSUE TITLE
Is it safe to use?

### ISSUE THREAD

**Issue by VladRafli** [felixfilipi/BOT-Anime-Downloader-RPA#1]
Hello, I'm stumbled upon your repository and read your README, when looking through the program design, there is something bothering me...

1. When searching an anime, is the top result is always the correct result?
2. When trying to download, is the download link correct? (Usually, website that provide animes are embeding redirection links that not always go to desired download link)
3. Is this handle download link that using url shortener?
4. How to ensure the downloaded file is correct? (Video content, file extension)
5. How this program know when anime batch available or not?
6. If the download fails on top result link, is it searches to another search result?

Maybe this repo need more explanation and having a worth of documentation so me as an user can assured its safety to use this.

Thank you!

**Comment by felixfilipi** [comment]
Hai Rafli, Sorry for the inconvenience. I believe that there's a lot of questions here because the documentation is not completed yet. To avoid this I will create and put the demo link in this repository as soon as I can to make it clear about the use of this repo.  Here's some answer due to your several questions.

1. I found that there's a miss in the activity label name which will be repaired soon. The process work by invoking your input to the google search, to find the web location which in my case using <a href="https://otakudesu.moe>/">Otakudesu download site</a>. Later on, the ROBOT will find the site according to the website name. So, the ROBOT didn't click only the top (1st) result but click the download site. Here's the example. 

<img src="https://i.ibb.co/c3K0P3R/issue1.png">

2. Yes it is, for the BATCH download case, the process will be fully automated until you click the download button. But for the 1 episode download case, you will be redirected only to the zippyshare download page. The redirection links will be blocked and closed but it takes some time for the ROBOT to find its element. But when I try to block the redirection links in the zippyshare, it takes a long time because the redirection link appeared a lot. So to avoid this, the ROBOT will take you only to the zippyshare download page for the 1 episode download case. But, I will try to find the solution to make it fully automated in the next repo version.
3. For this version, there's no URL shortener case in the download process. Because the ROBOT will find the anime only on otakudesu download page. 
4. The video extension has been set on the official otakudesu download page, so the requested file will be in accordance with the user's request with the file extension in the form of mp4. To ensure that the video content is correct, the ROBOT will compare the result string with the input from the user. So, there will be no miss video content happen.
<img src="https://i.ibb.co/FHD9kDj/issue2.png">

5. The ROBOT will keep running the process, and look for the BATCH element. If the requested element is not found (return FALSE on find element), the ROBOT will return the result as not found.
6. Unfortunately, for the 1st version, the downloading process just happen on the otakudesu download page. But, for the next version, it will be able to search it on another download page such as kusonime, and etc.

**Comment by VladRafli** [comment]
Thank you for the update, i'll wait until the documentation finished before using it.

And maybe last question, can i run this without downloading uipath by myself?

Thank you!

**Comment by felixfilipi** [comment]
Yes, I'm trying to figure that out. I think I will try to dockerizing the UiPath Orchestrator or somehow. But still trying tho :grin:

**Comment by felixfilipi** [comment]
The demo video has been uploaded to the documentation. Thx for asking. :smile:

**Comment by VladRafli** [comment]
Thank you for the demo, it should be useful guide to get started.

But its better to have guided text documentation to read as reading is much faster than watching video.

Cheers...
